doc/contribute/bin_blobs.rst

```diff
@@ -0,0 +1,229 @@
+.. _bin-blobs:
```

Comment:
I can't really tell who this document is for. It's trying to do too many things to live in the contribution guide in my opinion. If not now, then eventually I think we need to tease this apart a bit more.

Given that it's in the contribution guide, I guess the intended audience is people who want to submit blobs.

If that's the case, I think that the language could be tightened up a bit and reworked into more of a "here is how to submit blobs", "here are the things you can do", "here are the things you cannot do", "here are the things you must do".

Right now the language is kind of a mix between things like this:

> Zephyr supports downloading and using third-party binary blobs via its built-in mechanisms, with some important caveats, described in the following sections.

which reads like part of a user guide for how to use binary blobs, rather than a contributor guide for how to submit them, and stuff like this:

> It is up to the vendor to specify the license as part of the blob submission process.

Which is language I would expect to see in a contribution guide for blobs, and things like this:

> The Zephyr Project is not expected to be responsible for the maintenance and
> support of contributed binary blobs. As a consequence, at the discretion of the
> Zephyr Project release team, and on a case-by-case basis:
>
> - GitHub issues reported on the zephyr repository tracker that require use of
>   blobs to reproduce may not be treated as bugs

Which I would expect somewhere under "Project and Governance" in our documentation.

Reply:
I had the same feeling when writing it. But I looked at our current documentation for [external components](https://docs.zephyrproject.org/latest/contribute/external.html) and got some inspiration from the layout and contents in there. I would define this page as "the ultimate guide to binary blobs in Zephyr, for interested users and vendors alike". I don't necessarily think this is a bad thing: it combines all the info related to binary blobs and Zephyr in one single page, just like the external contributions does. However, some of this may end up in the modules page in the future, especially when I submit the next PR that contains the actual command and its documentation.

Reply:
>  I would define this page as "the ultimate guide to binary blobs in Zephyr, for interested users and vendors alike".

That is fine, but then I suggest placing it under doc/develop instead of doc/contribute, since the latter is for documentation regarding contributing to the project, while the former is a more general place for information about how to develop with zephyr.